test-programs/rnd-inf-test/tests/test.rs

```diff
@@ -0,0 +1,932 @@
+            &mut self.context,
+            &merkle_tree_keypair,
+            &nullifier_queue_keypair,
+            None,
+            1,
+        )
+        .await;
+        let merkle_tree = Box::new(light_merkle_tree_reference::MerkleTree::<Poseidon>::new(
+            STATE_MERKLE_TREE_HEIGHT as usize,
+            STATE_MERKLE_TREE_CANOPY_DEPTH as usize,
+        ));
+        self.indexer.state_merkle_trees.push(StateMerkleTreeBundle {
+            accounts: StateMerkleTreeAccounts {
+                merkle_tree: merkle_tree_keypair.pubkey(),
+                nullifier_queue: nullifier_queue_keypair.pubkey(),
+                cpi_context: Pubkey::new_unique(),
+            },
+            merkle_tree,
+        });
+        // TODO: Add assert
+    }
+
+    async fn create_address_tree(&mut self) {
+        let merkle_tree_keypair = Keypair::new(); //from_seed(&[self.rng.gen_range(0..255); 32]).unwrap();
+        let nullifier_queue_keypair = Keypair::new(); //from_seed(&[self.rng.gen_range(0..255); 32]).unwrap();
+        create_address_merkle_tree_and_queue_account(
+            &self.payer,
+            &mut self.context,
+            &merkle_tree_keypair,
+            &nullifier_queue_keypair,
+            None,
+            self.indexer.address_merkle_trees.len() as u64,
+        )
+        .await;
+        let init_value = BigUint::from_str_radix(
+            &"21888242871839275222246405745257275088548364400416034343698204186575808495616",
+            10,
+        )
+        .unwrap();
+        let mut merkle_tree = Box::new(
+            IndexedMerkleTree::<Poseidon, usize>::new(
+                STATE_MERKLE_TREE_HEIGHT as usize,
+                STATE_MERKLE_TREE_CANOPY_DEPTH as usize,
+            )
+            .unwrap(),
+        );
+        let mut indexed_array =
+            Box::new(IndexedArray::<Poseidon, usize, INDEXED_ARRAY_SIZE>::default());
+
+        merkle_tree.append(&init_value, &mut indexed_array).unwrap();
+        self.indexer
+            .address_merkle_trees
+            .push(AddressMerkleTreeBundle {
+                accounts: AddressMerkleTreeAccounts {
+                    merkle_tree: merkle_tree_keypair.pubkey(),
+                    queue: nullifier_queue_keypair.pubkey(),
+                },
+                merkle_tree,
+                indexed_array,
+            });
+        // TODO: Add assert
+    }
+
+    pub fn safe_gen_range<T, R>(rng: &mut StdRng, range: R, empty_fallback: T) -> T
+    where
+        T: SampleUniform + Copy,
+        R: SampleRange<T> + Sized,
+    {
+        if range.is_empty() {
+            return empty_fallback;
+        }
+        rng.gen_range(range)
+    }
+
+    /// 1. Transfer spl tokens between random users
+    pub async fn activate_keypair_actions(&mut self, user: &Pubkey) {
+        let user_index = self
+            .users
+            .iter()
+            .position(|u| &u.keypair.pubkey() == user)
+            .unwrap();
+        // compress spl
```

Comment:
Do you think we could move each operation - `compress spl`, `decompress spl`, `transfer spl`, `create address`, `compress sol`, `decompress sol`, `transfer sol` - to separate methods? From a brief look I can't see any of them sharing too much variables. And it would make easier for others to figure out the sequence of actions.